docs/rfcs/004-multi-file-diff.md

```diff
@@ -0,0 +1,89 @@
+# Commit Preview & Multi-file Diffs
+
+## :tipping_hand_woman: Status
+
+Proposed
+
+## :memo: Summary
+
+Give users an option to, before they make a commit, see diffs of all staged changes in one view, akin to the [`Files changed` tab in pull requests on github.com](https://github.com/atom/github/pull/1753/files).
+
+## :checkered_flag: Motivation
+
+So that users can view a full set of changes with more context before committing them.
+
+Note that the multi-diff view is the MVP of this RFC, and we have identified `Commit Preview` to be the least frictional way to introduce this feature without making too many UX changes. Other planned features that will also make use of multi-diff view are:
+
+- [commit pane item](https://github.com/atom/github/issues/1655) where it shows all changes in a single commit
+- [new PR review flow](https://github.com/atom/github/blob/master/docs/rfcs/003-pull-request-review.md) that shows all changed files proposed in a PR
+- (TBD) multi-select files from [unstaged](https://user-images.githubusercontent.com/378023/47553710-b60a5700-d942-11e8-8663-731b26d513c4.png) & [staged](https://user-images.githubusercontent.com/378023/47555145-0636e880-d946-11e8-85a7-f825278cc168.png) panes to view diffs
+
+## 🤯 Explanation
+
+#### Commit preview button
+A new button added above the commit message box that, when clicked, opens a multi-file diff pane item called something like "Commit Preview" and shows a summary of what will go into the user's next commit based on what is currently staged.
+
+![commit preview button](https://user-images.githubusercontent.com/378023/47554979-afc9aa00-d945-11e8-9953-45925e3278b9.png)
+
+#### Multi-file diff view
+
+![commit preview](https://user-images.githubusercontent.com/378023/47555097-e6072980-d945-11e8-9c29-05624825d9f8.png)
+
+- Shows diffs of multiple files as a stack.
+- Each diff retains the file-specific controls it currently has in its header (e.g. the open file, stage file, undo last discard, etc).
```

Comment:
In the mockups I changed it a bit to:

Description | Mockup
--- | ---
Open a file by clicking on the left of a hunk header (`<>` icon). It open the file, but also scrolls to the appropriate line. | ![open file](https://user-images.githubusercontent.com/378023/47560751-22da1d00-d954-11e8-919a-0c1e7e369dc4.png)
Unstage the entire file or individual hunks by clicking on the up arrow icons. | ![image](https://user-images.githubusercontent.com/378023/47560984-f07cef80-d954-11e8-99a2-c4229517b7e2.png)
When selecting lines, an additional "Unstage Selected Lines" button is shown. | ![image](https://user-images.githubusercontent.com/378023/47561162-7436dc00-d955-11e8-9460-1faf0a02f43f.png)

For the Commit Preview, there is no discard button, since I _think_ discarding is only possible for unstaged changes.

The above changes could easily be moved to "Out of scope" and considered at a later point. Then we can just re-use what we currently have for each file:

![image](https://user-images.githubusercontent.com/378023/47561459-469e6280-d956-11e8-90d2-50f37e509ee2.png)

Reply:
>  I think discarding is only possible for unstaged changes.

This is true 👍